component/src/test/java/org/wso2/extension/siddhi/execution/ml/classification/perceptron/PerceptronClassifierStreamProcessorExtensionTestCase.java

```diff
@@ -0,0 +1,516 @@
+                + trainingQuery + query);
+        siddhiAppRuntime.addCallback("query1", new QueryCallback() {
+
+            @Override
+            public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                count++;
+                if (count == 1) {
+                    AssertJUnit.assertArrayEquals(new Object[]{0.8, 0.67, 0.1, 0.03, true}, inEvents[0].getData());
+                } else if (count == 2) {
+                    AssertJUnit.assertArrayEquals(new Object[]{0.33, 0.23, 0.632, 0.992, false}, inEvents[0].getData());
+                } else if (count == 3) {
+                    AssertJUnit.assertArrayEquals(new Object[]{0.8, 0.67, 0.1, 0.03, false}, inEvents[0].getData());
+                } else if (count == 4) {
+                    AssertJUnit.assertArrayEquals(new Object[]{0.33, 0.23, 0.632, 0.992, true}, inEvents[0].getData());
+                }
+            }
+        });
+        try {
+            InputHandler inputHandler = siddhiAppRuntime.getInputHandler("StreamTrain");
+            siddhiAppRuntime.start();
+            inputHandler.send(new Object[]{1.0, 1.0, 0.2, 0.13, "true"});
+            inputHandler.send(new Object[]{0.9, 0.89, 0.3, 0.02, "true"});
+            inputHandler.send(new Object[]{0.0, 0.0, 1.0, 0.82, "false"});
+            inputHandler.send(new Object[]{0.01, 0.4, 0.77, 0.92, "false"});
+            inputHandler.send(new Object[]{0.80, 0.81, 0.11, 0.13, "true"});
+            inputHandler.send(new Object[]{0.02, 0.30, 0.88, 0.76, "false"});
+            inputHandler.send(new Object[]{0.93, 0.71, 0.02, 0.122, "true"});
+            inputHandler.send(new Object[]{0.29, 0.24, 0.98, 0.65, "false"});
+            Thread.sleep(1100);
+
+            InputHandler inputHandler1 = siddhiAppRuntime.getInputHandler("StreamA");
+            // send some unseen data for prediction
+            inputHandler1.send(new Object[]{0.8, 0.67, 0.1, 0.03});
+            inputHandler1.send(new Object[]{0.33, 0.23, 0.632, 0.992});
+            Thread.sleep(1000);
+            // try to drift the model
+            inputHandler.send(new Object[]{0.88, 1.0, 0.2, 0.13, "false"});
+            inputHandler.send(new Object[]{0.9, 0.79, 0.3, 0.02, "false"});
+            inputHandler.send(new Object[]{0.0, 0.0, 0.90, 0.72, "true"});
+            inputHandler.send(new Object[]{0.01, 0.4, 0.87, 0.62, "true"});
+            inputHandler.send(new Object[]{0.90, 0.91, 0.11, 0.13, "false"});
+            inputHandler.send(new Object[]{0.02, 0.30, 0.88, 0.66, "true"});
+            inputHandler.send(new Object[]{0.83, 0.79, 0.02, 0.122, "false"});
+            inputHandler.send(new Object[]{0.29, 0.24, 0.98, 0.77, "true"});
+            Thread.sleep(1000);
+            // send some unseen data for prediction
+            inputHandler1.send(new Object[]{0.8, 0.67, 0.1, 0.03});
+            inputHandler1.send(new Object[]{0.33, 0.23, 0.632, 0.992});
+
+            Thread.sleep(1100);
```

Comment:
Have an assertion for the total received events at the end, this is because if no output is generated still this test will pass, add similar checks for all test cases